src/util/hashFetch/expoRetry.js

```diff
@@ -0,0 +1,118 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+const fs = window.require('fs');
+const util = window.require('util');
+const path = window.require('path');
+
+import { readJson as fsReadJson, writeJson as fsWriteJson } from 'fs-extra';
+const fsExists = util.promisify(fs.stat);
+
+import { getHashFetchPath } from '../host';
+
+// Handle retries with exponential delay for failed download attempts
+export default class ExpoRetry {
+  static instance = null;
+
+  // We store the URL to allow GH Hint URL updates for a given hash
+  // to take effect immediately, and to disregard past failed attempts at
+  // getting this file from other URLs.
```

Comment:
Should we? It's a DDoS vector if the target host is the same. If the attacker registers `(some hash, "http://victim-website.com/")` on GH Hint and then every few minutes changes the URL to `http://victim-website.com/?x` `http://victim-website.com/?xx` `http://victim-website.com/?xxx` then ExpoRetry isn't used at all..
Or is this a non-issue because of gas costs?

I'm thinking in particular if somebody registers a dapp with the icon or manifest URL in GitHub Hint set to some big file like `http://victim-website.com/bigfile.zip`
The icons and manifests of network dapps are loaded each time Parity UI users go to the homepage.
Amount of Parity UI users × Amount of times they go to the homepage × The attacker regularly updates the URL to `http://victim-website.com/bigfile.zip?x` etc. = victim website has a bad time

This could be fixed if we key the attempts based on the hash only. What do you think?

Reply:
cc @tomusdrw